misc/synchronization.md

```diff
@@ -0,0 +1,413 @@
+// Task1 - Task2 - Task4 ~ Task6
+//
+// Task3 - Task5 ~ Task7
+//
+// Note the that the ~ dependencies include the call to the destructor of mv.
+```
+- Example 2: Basic Usage with `done()`.
+```cpp
+  Matrix m;
+
+  hpx::dataflow(Task1, m({0, 0}));
+  hpx::dataflow(Task2, m({0, 0}));  // Depends on Task1.
+  hpx::dataflow(Task3, m({0, 1}));  // Different tile. No dependency on Task1 or Task2.
+
+  MatrixView mv(UpLo::General, m);
+
+  hpx::dataflow(Task4, m({0, 0}));   // Depends on done notification and Task6
+  hpx::dataflow(Task5, m({0, 1}));   // Depends on done notification and Task7
+  hpx::dataflow(Task6, mv({0, 0}));  // Depends on Task2.
+  hpx::dataflow(Task7, mv({0, 1}));  // Depends on Task3.
+
+  mv.done({0, 0});
+  mv.done({0, 1});
+
+// Resulting dependency graph:
+// Task1 - Task2 - Task6 ~ Task4
+//
+// Task3 - Task7 ~ Task5
+//
+// Note the that the ~ dependencies include the dependency on the call to done().
+```
+- Example 3: Read cuncurrency.
+```cpp
+  Matrix m;
+
+  hpx::dataflow(Task1, m({0, 0}));
+  hpx::dataflow(Task2, m.read({0, 0}));  // Depends on Task1
+
+  MatrixView mv(UpLo::General, m);
+
+  hpx::dataflow(Task3, m.read({0, 0}));   // Depends on doneWrite notification and Task5
+  hpx::dataflow(Task4, mv.read({0, 0}));  // Depends on Task2.
+  hpx::dataflow(Task5, mv({0, 0}));       // Depends on Task2 and Task4.
+
+  mv.doneWrite({0, 0});
+
+  hpx::dataflow(Task6, mv.read({0, 0}));  // Depends on Task5.
+
+  mv.done({0, 0});
+
+  hpx::dataflow(Task7, m.read({0, 0}));  // Depends on doneWrite notification and Task5
+  hpx::dataflow(Task8, m({0, 0}));       // Depends on done notification and Task3,6,7
+
+// Resulting dependency graph:
+// Task1 - Task2 - Task5 -  Task6 -= Task8
+//       \ Task4 /       \~ Task3 /
+//                       \~ Task7 /
+//
+// Note the that the \~ dependencies include the dependency on the call to doneWrite().
+// Note the that the -= dependencies include the dependency on the call to done().
+```
+
+## Matrix and Matrix Tiles advanced: Tile dependencies mechanism
+
+The dependency are managed with a promise-future mechanism.
+
+When the destructor of a tile obtained with `operator()` or `read()` is called, the value
+of the next future is set using the promise which was assigned by the `Matrix` object.
+Note that the destructor of the tile contained in the shared future is only called when all
+instances of the shared futures go out of scope, therefore,
+to avoid deadlocks the scope of tile shared futures needs extra care.
+
+Note that returning matrix tiles from tasks may be a source of deadlocks,
+therefore it should be avoided if possible. Otherwise a carefull analysis of the dependencies is needed to avoid problems.
+
+### Examples advanced use
+
+(Note: this examples are used to illustrate how dependencies are determined,
+however they cannot be compiled since they are simplified to improve readability.
+(e.g. constructors and template parameters are omitted)).
```

Comment:
```suggestion
however they cannot be compiled since they are simplified to improve readability
(e.g. constructors and template parameters are omitted).)
```